inc/composer/class-command.php

```diff
@@ -0,0 +1,195 @@
+	 */
+	protected function execute( InputInterface $input, OutputInterface $output ) {
+		$subcommand = $input->getArgument( 'subcommand' );
+		switch ( $subcommand ) {
+			case 'phpunit':
+				return $this->run_command( $input, $output, 'vendor/bin/phpunit' );
+
+			case 'scaffold':
+				return $this->scaffold( $input, $output );
+
+			default:
+				throw new CommandNotFoundException( sprintf( 'Subcommand "%s" is not defined.', $subcommand ) );
+		}
+	}
+
+	/**
+	 * Copies boilerplate files over to the project root.
+	 *
+	 * @param InputInterface $input
+	 * @param OutputInterface $output
+	 * @return int
+	 */
+	protected function scaffold( InputInterface $input, OutputInterface $output ) {
+		// Scaffold PHPUnit by default.
+		$target = $input->getArgument( 'options' )[0] ?? null;
+
+		switch ( $target ) {
+			case 'phpunit':
+				return $this->scaffold_phpunit( $input, $output );
+
+			case 'travis':
+				return $this->scaffold_travis( $input, $output );
+
+			default:
+				throw new CommandNotFoundException( sprintf(
+					'"%s" is not a recognised scaffold type, it must be one of "phpunit" or "travis".',
+					$target
+				) );
+		}
+	}
+
+	protected function scaffold_phpunit( InputInterface $input, OutputInterface $output ) {
+		$package_root = dirname( __DIR__, 2 );
+
+		// Check for an existing tests directory.
+		if ( file_exists( $this->get_root_dir() . '/.tests' ) ) {
+			$output->writeln( '<error>You already have a ".tests" directory present in your project!</>' );
+			$output->writeln( '<comment>If you want to compare the recommended Altis testing setup you can temporarily rename your current tests directory and run this command again.</>' );
+			return 1;
+		}
+
+		if ( file_exists( $this->get_root_dir() . '/phpunit.xml.dist' ) ) {
+			$output->writeln( '<error>phpunit.xml.dist file found, skipping</>' );
+			$output->writeln( '<comment>You should ensure your version of the file uses the same values shown in this basic outline:</>' );
+			$output->write(
+				'<phpunit bootstrap=".tests/bootstrap.php">' . "\n" .
+				"\t" . '<php>' . "\n" .
+				"\t\t" . '<env name="WP_PHPUNIT__TESTS_CONFIG" value=".tests/config.php" />' . "\n" .
+				"\t\t" . '<const name="WP_PHP_BINARY" value="/usr/bin/env php" />' . "\n" .
+				"\t" . '</php>' . "\n" .
+				'</phpunit>'
+			);
+		} else {
+			$output->writeln( '<info>Copying phpunit.xml.dist</>' );
+			copy( $package_root . '/boilerplate/phpunit.xml.dist', $this->get_root_dir() . '/phpunit.xml.dist' );
+		}
+
+		// Copy files over.
+		$output->writeln( '<info>Creating .tests directory...</>' );
+		mkdir( $this->get_root_dir() . '/.tests/inc', 0755, true );
+		$output->writeln( '<info>Copying bootstrap.php</>' );
+		copy( $package_root . '/boilerplate/bootstrap.php', $this->get_root_dir() . '/.tests/bootstrap.php' );
+		$output->writeln( '<info>Copying config.php</>' );
+		copy( $package_root . '/boilerplate/config.php', $this->get_root_dir() . '/.tests/config.php' );
```

Comment:
Is there any way we can not have these as boilerplate? If possible I think we should start as zero-config and expand as we go / as it's required.

Reply:
I did wonder about that but it's difficult without sacrificing some flexibility I think. A lot of our projects customise these files quite considerably but I'll look into it.